internal/service/handlers/activate_balance.go

```diff
@@ -95,33 +98,16 @@ func ActivateBalance(w http.ResponseWriter, r *http.Request) {
 			return fmt.Errorf("failed to consume referral: %w", err)
 		}
 
-		if balance.IsVerified {
-			// Be referred event is a welcome bonus when you created balance with non-genesis referral code
-			if err = claimBeReferredEvent(r, *balance); err != nil {
-				return fmt.Errorf("failed to claim be referred event: %w", err)
-			}
-		}
-
-		evTypeRef := EventTypes(r).Get(models.TypeReferralSpecific, evtypes.FilterInactive)
-		if evTypeRef == nil {
-			Log(r).Debug("Referral specific event type is inactive")
+		if !balance.IsVerified {
+			log.Debug("Balance is not verified, events will not be claimed")
 			return nil
 		}
 
-		if balance.IsVerified {
-			if err = claimReferralSpecificEvents(r, *evTypeRef, balance.Nullifier); err != nil {
-				return fmt.Errorf("failed to claim referral specific events: %w", err)
-			}
-		}
-
-		if err = addEventForReferrer(r, *evTypeRef, *balance); err != nil {
-			return fmt.Errorf("add event for referrer: %w", err)
-		}
-
-		return nil
+		balance.ReferredBy = &referral.ID
+		return doVerificationEventUpdates(r, *balance)
```

Comment:
We need to add referral specific event for referrer even if referre doesn't scan passport

Reply:
no, we don't :) (logic resolved internally)